Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Services/FileShareServiceCacheTest.cs

```diff
@@ -0,0 +1,176 @@
+
+            A.CallTo(() => _azureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns("testConnectionString");
+
+            A.CallTo(() => _azureTableStorageClient.GetEntityAsync(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored))
+                           .Returns(customTableEntity);
+
+            A.CallTo(() => _azureTableStorageClient.DeleteEntityAsync(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored))
+                          .MustNotHaveHappened();
+
+            BatchSearchResponseModel result = await _fileShareServiceCache.GetWeeklyBatchFilesFromCache(2022, 2, string.Empty);
+
+            Assert.IsInstanceOf<BatchSearchResponseModel>(result);
+            Assert.IsNull(result.batchSearchResponse);
+        }
+
+        [Test]
+        public async Task WhenFSSCacheCallsGetWeeklyBatchFilesFromCache_ThenReturnsBatchSearchResponse()
+        {
+            BatchSearchResponseModel batchSearchResponseModel = new();
+            CustomTableEntity customTableEntity = new()
+            {
+                PartitionKey = "Public",
+                RowKey = "",
+                Response = "{\"count\":2,\"total\":2,\"entries\":[{\"batchId\":\"e637f124-27bb-4f6a-a413-3b3f879725cf\",\"status\":\"Committed\",\"attributes\":[{\"key\":\"Content\",\"value\":\"tracings\"}],\"businessUnit\":\"MaritimeSafetyInformation\",\"batchPublishedDate\":\"2022-07-06T12:07:23.137Z\",\"expiryDate\":\"2023-01-09T00:00:00Z\",\"files\":[{\"filename\":\"rs6-nms-2020-01.xml\",\"fileSize\":8758,\"mimeType\":\"application/xml\",\"hash\":\"mWZtPqCbWAqAqT4KPHV83w==\",\"attributes\":[],\"links\":{\"get\":{\"href\":\"/batch/e637f124-27bb-4f6a-a413-3b3f879725cf/files/rs6-nms-2020-01.xml\"}}}],\"allFilesZipSize\":3605391}],\"_links\":{ \"self\":{ \"href\":\"/batch?limit=100\"} }}"
+            };
+
+            customTableEntity.CacheExpiry = DateTime.UtcNow.AddMinutes(2);
+            batchSearchResponseModel.batchSearchResponse = JsonConvert.DeserializeObject<BatchSearchResponse>(customTableEntity.Response);
+
+            A.CallTo(() => _azureStorageService.GetStorageAccountConnectionString(A<string>.Ignored, A<string>.Ignored)).Returns("testConnectionString");
+
+            A.CallTo(() => _azureTableStorageClient.GetEntityAsync(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored))
+                           .Returns(customTableEntity);
+
+            BatchSearchResponseModel result = await _fileShareServiceCache.GetWeeklyBatchFilesFromCache(2022, 2, string.Empty);
+
+            Assert.IsInstanceOf<BatchSearchResponseModel>(result);
+            Assert.AreEqual(batchSearchResponseModel.batchSearchResponse.Count, result.batchSearchResponse.Count);
+        }
+
+        [Test]
+        public async Task WhenFSSCacheCallsInsertEntityAsync_ThenReturnsBatchSearchResponse()
```

Comment:
Is this test finished?  It doesn't seem to test what the title implies, and doesn't assert very much,

Reply:
Thanks for highlighting, Renamed test title we checked InsertEntityAsync flow only.